server-spi/src/main/java/org/keycloak/services/clientpolicy/ClientPolicyEvent.java

```diff
@@ -21,6 +21,7 @@
 
     REGISTER,
     UPDATE,
+    DELETE,
```

Comment:
As https://github.com/tnorimat/keycloak/blob/37586e0ebf4ab1a4538fc8a7a85bc25258683b7f/server-spi/src/main/java/org/keycloak/services/clientpolicy/ClientPolicyEvent.java#L20-L34 shows, we might support 4 events (`REGISTERED` corresponding to `afterRegister` in the existing client registration policy, `UPDATED` corresponding to `afterUpdate` in the existing client registration policy , `VIEW` corresponding to `beforeView` in the existing client registration policy , `UNREGISTER` corresponding to `beforeDelete` in the existing client registration policy) that the existing client registration policy supported but the client policy has not yet supported.

There are two method to delete the client, via OIDC Dynamic Client Registration and via Admin REST API.
I infer that this PR tries to support deleting the client via Admin REST API. We might treat these 2 types of deleting the client event as the one `UNREGISTER` event because other events like `REGISTER` convers both cases via OIDC Dynamic Client Registration and via Admin REST API.